src/base/transport.ts

```diff
@@ -0,0 +1,130 @@
+import OAuth from 'oauth';
+import IClientOptions from './IClientOptions';
+import Cache from './Cache';
+import { formatURL } from './utils';
+
+class Transport {
+  private oauth: OAuth.OAuth;
+
+  private cache?: Cache;
+  private credentials: IClientOptions & { [key: string]: any };
+  constructor(options: IClientOptions) {
+    this.credentials = options;
+    this.oauth = new OAuth.OAuth(
+      'https://api.twitter.com/oauth/request_token',
+      'https://api.twitter.com/oauth/access_token',
+      this.credentials.apiKey,
+      this.credentials.apiSecret,
+      '1.0A',
+      null,
+      'HMAC-SHA1'
+    );
+    if (!options?.disableCache) {
+      this.cache = new Cache(options?.ttl, options.maxByteSize);
+    }
+  }
+  updateOptions(options: Partial<IClientOptions>) {
```

Comment:
Same here - let's have an extra empty line before each new method :blush: 

Also, to stick with the convention of the rest of the codebase, let's keep a `public` access modifier.
```suggestion
  public updateOptions(options: Partial<IClientOptions>) {
```

Reply:
Done - add space before each method and labels them as public.